src/dimension/Dimension.js

```diff
@@ -0,0 +1,488 @@
+        // todo - temporary hack to support backwards compatibility
+        this.remove = this.dispose  = () => this.getCrossfilter().removeDimension(this)
+        this.value                  = () => this._dimArray
+        this.getCrossfilterId       = () => crossfilter.getId()
+        this.getProjectOn           = () => this._projectExpressions
+        this.setDrillDownFilter     = (value) => {
+            this._drillDownFilter = value
+            return this
+        }
+        this.getTable               = crossfilter.getTable
+        // testing haxx
+        this.getSamplingRatio       = () => this._samplingRatio
+    }
+    set(fn) {
+        this._dimArray = fn(this._dimArray)
+        return this
+    }
+    /******************************************************************
+     * private methods
+     */
+    /******************************************************************
+     * public methods
+     */
+    // todo - tricky: look at immerse/src/services/crossfilter.getTopN
+    group(groupId = false) { // todo - shouldn't this at least be parameterized like thus?
+        // if(!this._dimensionGroups.length) {
+        //     const newGroup = new Group(this.getDataConnector(), this)
+        //     return this.addGroupToDimension(newGroup)
+        // }
+        // else {
+        //     // todo - such an excellent question, eh? (tisws)
+        // }
+        const newGroup = new Group(this.getDataConnector(), this)
+        return this.addGroupToDimension(newGroup)
+    }
+    addGroupToDimension(newGroup) {
+        this._dimensionGroups.push(newGroup)
+        return newGroup
+    }
+    groupAll() {
+        return this.getCrossfilter().groupAll
+    }
+    setEliminateNull(eliminateNull) {
+        this._eliminateNull = eliminateNull
+        return this
+    }
+    getEliminateNull() {
+        return this._eliminateNull
+    }
+    /**
+     *  tbd public or private methods
+     */
+    multiDim (value) {
+        if (typeof value === "boolean") {
+            this._isMultiDim = value
+            return this
+        }
+        return this._isMultiDim
+    }
+    // todo - make this consistent with immerse/src/services/crossfilter.getTopN(),
+    // todo - which passes in 'column'
+    order(orderExpression) {
+        this._orderExpression = orderExpression
+        return this
+    }
+    orderNatural() {
+        this._orderExpression = null
+        return this
+    }
+    allowTargeted(allowTargeted) {
+        if (!arguments.length) {
+            return this._allowTargeted
+        }
+        this._allowTargeted = allowTargeted
+        return this // todo - returning 'this' is inconsistent
+    }
+    toggleTarget() {
+        const crossfilter = this.getCrossfilter()
+        if (crossfilter._targetFilter == this._dimensionIndex) { // TODO duplicates isTargeting
+            crossfilter._targetFilter = null // TODO duplicates removeTarget
```

Comment:
for these two `TODO`s
better to be explicit with `===`
looks like isTargeting removeTarget and toggleTarget are all special cases of a more general function. Maybe something like this?
```js
_targeter (targetFilterFunc) { targetFilterFunc(this.getCrossfilter()._targetFilter) }
removeTarget() { return _targeter(targetFilter => targetFilter = null) }
isTargeting() { return _targeter(targetFilter => targetFilter === this._dimensionIndex) }
toggleTarget() { return _targeter(targetFilter => targetFilter = this._dimensionIndex || null) }
```